arch/arm/core/cortex_r/reset.S

```diff
@@ -99,4 +100,8 @@ SECTION_SUBSEC_FUNC(TEXT,_reset_section,__start)
 	bl z_arm_watchdog_init
 #endif
 
+#if defined(CONFIG_DISABLE_TCM_ECC)
+	bl disable_ecc
```

Comment:
Is this something that is SoC specific or ARCH specific?
If SoC specific, I would suggesting doing this under PLATFORM_SPECIFIC_INIT (see for example Cortex-M reset.S)

Reply:
This should be at arch-level. TCM ECC is a feature that is directly integrated into the Cortex-R processors.